src/entropic/greenkhorn.jl

```diff
@@ -0,0 +1,114 @@
+# Greenkhorn is a greedy version of the Sinkhorn algorithm
+# This method is from https://arxiv.org/pdf/1705.09634.pdf
+# Code is based on implementation from package POT
+
+
+"""
+    Greenkhorn()
+
+Greenkhorn is a greedy version of the Sinkhorn algorithm.
+"""
+struct Greenkhorn <: Sinkhorn end
+
+struct GreenkhornCache{U,V,KT}
+    u::U
+    v::V
+    K::KT
+    Kv::U #placeholder
+    G::KT
+    du::U
+    dv::V
+end
+
+Base.show(io::IO, ::Greenkhorn) = print(io, "Greenkhorn algorithm")
+
+function build_cache(
+    ::Type{T},
+    ::Greenkhorn,
+    size2::Tuple,
+    μ::AbstractVecOrMat,
+    ν::AbstractVecOrMat,
+    C::AbstractMatrix,
+    ε::Real,
+) where {T}
+    # compute Gibbs kernel (has to be mutable for ε-scaling algorithm)
+    K = similar(C, T)
+    @. K = exp(-C / ε)
+
+    # create and initialize dual potentials
+    u = similar(μ, T, size(μ, 1), size2...)
+    v = similar(ν, T, size(ν, 1), size2...)
+    fill!(u, one(T)/size(μ, 1))
+    fill!(v, one(T)/size(ν, 1))
+
+    G = sinkhorn_plan(u, v, K)
+    # G = diagm(u) * K * diagm(v)
+
+    Kv = similar(u)
+
+    # This is me triying to get the `batch tests to work`
+    # improve this!
+    # if (length(size(μ)) == 2 && length(size(ν)) == 1)
+    #     du = reshape(sum(G, dims=2), size(μ)) - μ
+    #     dv = reshape(sum(G, dims=1),size(v)) - repeat(ν,1,size(v)[2])
+    # elseif (length(size(μ)) == 1 && length(size(ν)) == 2)
+    #     du = reshape(sum(G, dims=2),size(u)) - repeat(μ,1,size(u)[2])
+    #     dv = reshape(sum(G, dims=1), size(ν)) - ν
+    # else
+        du = reshape(sum(G, dims=2), size(μ)) - μ
+        dv = reshape(sum(G, dims=1), size(ν)) - ν
+    # end
+
+
+    return GreenkhornCache(u, v, K, Kv, G, du, dv)
+end
+
+prestep!(::SinkhornSolver{Greenkhorn}, ::Int) = nothing
+
+init_step!(solver::SinkhornSolver{<:Greenkhorn}) = nothing
+
+function step!(solver::SinkhornSolver{<:Greenkhorn}, iter::Int)
+    μ = solver.source
+    ν = solver.target
+    cache = solver.cache
+    u = cache.u
+    v = cache.v
+    K = cache.K
+    G = cache.G
+    Δμ= cache.du
+    Δν= cache.dv
+
+    # The implementation in POT does not compute `μ .* log.(μ ./ sum(G', dims=1)[:])`
+    # or `ν .* log.(ν ./ sum(G', dims=2)[:])`. Yet, this term is present in the original
+    # paper, where it uses ρ(a,b) = b - a + a log(a/b).
+    # ρμ = abs.(Δμ + μ .* log.(μ ./ sum(G', dims=1)[:]))
+    # ρν = abs.(Δν + ν .* log.(ν ./ sum(G', dims=2)[:]))
+    
+    i₁ = argmax(abs.(Δμ))
+    i₂ = argmax(abs.(Δν))
+
+    # if ρμ[i₁]> ρν[i₂]
+    if abs(Δμ[i₁]) > abs(Δν[i₂])
+        old_u = u[i₁]
```

Comment:
Also has to be changed for batch support it seems.
```suggestion
        old_u = u[Δμ_max_idx]
```